src/main/java/org/mule/extension/validation/internal/privileged/AggregateOperationExecutor.java

```diff
@@ -64,10 +64,12 @@ public void execute(ExecutionContext<OperationModel> executionContext, ExecutorC
       final CoreEvent processEvent = CoreEvent.builder(childContext, event).build();
       try {
         // The chain is initialized with the muleContext so that it can run correctly
-        initialiseIfNeeded(messageChain, muleContext);
-        startIfNeeded(messageChain);
-        CoreEvent result = messageChain.process(processEvent);
-        childContext.success(result);
+        synchronized (processor) {
```

Comment:
@juanfloresMule please add a comment on why the synchronized block is using an asset that is not used in the block

Reply:
I commented on the reason why we used the synchronized block, and the reference to the research that gave rise to it.